vllm/v1/sample/sampler.py

```diff
@@ -114,12 +148,23 @@ def sample(
 
         assert not (sampling_metadata.all_greedy
                     and sampling_metadata.all_random)
+        return_processed_logits = (
+            sampling_metadata.max_num_logprobs is not None
+            and self.logprobs_mode == "processed_logits")
+        return_processed_logprobs = (
+            sampling_metadata.max_num_logprobs is not None
+            and self.logprobs_mode == "processed_logprobs")
+        processed_logprobs = None
         if sampling_metadata.all_random:
             greedy_sampled = None
         else:
             greedy_sampled = self.greedy_sample(logits)
             if sampling_metadata.all_greedy:
-                return greedy_sampled
+                if return_processed_logits:
+                    processed_logprobs = logits
+                elif return_processed_logprobs:
+                    processed_logprobs = self.compute_logprobs(logits)
+                return greedy_sampled, processed_logprobs
```

Comment:
This can now be simplified, the intermediate variables aren't needed and checks can all be moved within `if sampling_metadata.all_greedy:`